emcc.py

```diff
@@ -1467,7 +1432,7 @@ def check(input_file):
       cflags.append('-DEMSCRIPTEN')
 
     # Treat the empty extension as an executable, to handle the commond case of `emcc -o foo foo.c`
```

Comment:
please add a comment here

or maybe `('.out', '')` could be another constant, `OTHER_EXECUTABLE_ENDINGS`?